packages/url-loader/src/types/plugins.ts

```diff
@@ -9,14 +11,27 @@ export interface PluginSettings<Options extends AllOptions = AllOptions> {
   options: Options;
 }
 
-export interface PluginOverrides {
-  format?: string;
-  width?: number;
-}
-
 export interface TransformationPlugin {
   assetTypes: Array<string>;
   plugin: Function;
   strict?: boolean;
   props: object;
+}
+
+
+export const pluginOptionsSchema = z.object({
+  aspectRatio: aspectRatio.schema.optional(),
+  crop: crop.schema.optional(),
+  gravity: gravity.schema.optional(),
+  height: height.schema.optional(),
+  format: format.schema.optional(),
+  width: width.schema.optional(),
+  zoom: zoom.schema.optional(),
+})
+
+export type PluginOptions = z.infer<typeof pluginOptionsSchema>;
+
+export interface PluginResults {
+  options?: PluginOptions;
+  resize?: string;
```

Comment:
question: maybe I missed in the description but why this `resize` is here and not in the `pluginOptionsSchema`?

Reply:
my thought process here when doing this was that PluginOptions only includes first-class parameters, where resize is a mix of such. though in reality im not sure that the distinction actually matters here, unless in the future there's an angle for directly applying that object to options, but that's not currently the case

will move it inside since i think it probably makes sense and practically it won't make a difference with the current state of it

Reply:
definitely cleaner this way 👍